server/router.js

```diff
@@ -19,6 +21,8 @@ router.get('/category/:categoryId/food', foodCategory);
 router.get('/food', getFood);
 router.get('/user/calories', getUserCalories);
 
+router.post('/category/:categoryId/food/:foodId', insertFoodController);
```

Comment:
It's probably too late to change this, but it's best if create or insert APIs do not require the client to pass an object ID, and instead let the backend decide what the newly-created item's ID should be (usually just with an auto-increment id field in the table).
This allows the client to create a new item with this url: `POST '/category/:categoryId/food/`
and (most importantly) frees the client to use this url for modify an existing food item: `POST '/category/:categoryId/food/:foodId`

It also avoids the problem where the client is trying to decide what the new food item's ID should be, which can be problematic (what if two admins, at close to the same time, create a new food item and both of their client javascript tries to create it with the same ID?)

Reply:
actually, this route is not for inserting food into the database, it's just for adding the quantity provided by the user, so we need to send the food quantity

Reply:
Oh sorry, I misunderstood.  Thanks for letting me know.